### PR TITLE
enable None model checkpoint default

### PR DESCRIPTION
**Pull request by williamFalcon**
Since we don't know what to monitor, we are using None as the default going forward.

Now if a user wants to monitor a value, they have to specify a monitor value in the callback.

This is an intermediate step to getting rid of the magic keywords for checkpoint and ES

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -336,6 +340,13 @@ def on_validation_end(self, trainer, pl_module):
         metrics = trainer.logger_connector.callback_metrics
         epoch = trainer.current_epoch
 
+        # backward compatibility... need to deprecate
```
Comment (ananthsub) [inline pytorch_lightning/callbacks/model_checkpoint.py:343]: these need to check if `self.monitor is None` before resetting it to the older defaults right?

Comment (williamFalcon) [reply pytorch_lightning/callbacks/model_checkpoint.py:343]: yup! sorry, i think i removed the WIP too soon haha. but good catch

Comment (williamFalcon) [reply pytorch_lightning/callbacks/model_checkpoint.py:403]: ended up just doing a much needed clean up to maintain sanity haha

File: pytorch_lightning/trainer/trainer.py [hunk 2]
```diff
@@ -513,7 +513,6 @@ def run_sanity_check(self, ref_model):
             self.on_sanity_check_end()
             self.running_sanity_check = False
 
-    @trainer_state(entering=TrainerState.RUNNING, exiting=TrainerState.FINISHED)
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/trainer.py:516]: @Borda  let's please get rid of these... makes readability impossible (and IDE inspection)

Comment (Borda) [reply pytorch_lightning/trainer/trainer.py:516]: ok, we can remove wrappers, just keep the state assignment..

**Pull request merged**